src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1512 @@
+            DQ bitfield array should be *ignored* (i.e., zeroed).
+
+            When co-adding models using :py:meth:`add_model`, any pixels with
+            a non-zero DQ values are assigned a weight of zero and therefore
+            they do not contribute to the output (resampled) data.
+            ``good_bits`` provides a mean to ignore some of the DQ bitflags.
+
+            When ``good_bits`` is an integer, it must be
+            the sum of all the DQ bit values from the input model's
+            DQ array that should be considered "good" (or ignored). For
+            example, if pixels in the DQ array can be
+            combinations of 1, 2, 4, and 8 flags and one wants to consider DQ
+            "defects" having flags 2 and 4 as being acceptable, then
+            ``good_bits`` should be set to 2+4=6. Then a pixel with DQ values
+            2,4, or 6 will be considered a good pixel, while a pixel with
+            DQ value, e.g., 1+2=3, 4+8=12, etc. will be flagged as
+            a "bad" pixel.
+
+            Alternatively, when ``good_bits`` is a string, it can be a
+            comma-separated or '+' separated list of integer bit flags that
+            should be summed to obtain the final "good" bits. For example,
+            both "4,8" and "4+8" are equivalent to integer ``good_bits=12``.
+
+            Finally, instead of integers, ``good_bits`` can be a string of
+            comma-separated mnemonics. For example, for JWST, all the following
+            specifications are equivalent:
+
+            `"12" == "4+8" == "4, 8" == "JUMP_DET, DROPOUT"`
+
+            In order to "translate" mnemonic code to integer bit flags,
+            ``Resample.dq_flag_name_map`` attribute must be set to either
+            a dictionary (with keys being mnemonc codes and the values being
+            integer flags) or a `~astropy.nddata.BitFlagNameMap`.
+
+            In order to reverse the meaning of the flags
+            from indicating values of the "good" DQ flags
+            to indicating the "bad" DQ flags, prepend '~' to the string
+            value. For example, in order to exclude pixels with
+            DQ flags 4 and 8 for computations and to consider
+            as "good" all other pixels (regardless of their DQ flag),
+            use a value of ``~4+8``, or ``~4,8``. A string value of
+            ``~0`` would be equivalent to a setting of ``None``.
+
+            Default value (0) will make *all* pixels with non-zero DQ
+            values be considered "bad" pixels, and the corresponding data
+            pixels will be assigned zero weight and thus these pixels
+            will not contribute to the output resampled data array.
+
+            Set `good_bits` to `None` to turn off the use of model's DQ
+            array.
+
+            For more details, see documentation for
+            `astropy.nddata.bitmask.extend_bit_flag_map`.
+
+        output_wcs : dict, WCS object, None
```

Comment:
Why are there so many options for this (and the other input arguments)? Is there ever a place where `ouput_wcs` is not a dict?

Reply:
For example here: https://github.com/spacetelescope/jwst/blob/144c4dcf88df16d2f143a82c936d07fc77c7c13c/jwst/resample/resample_step.py#L179-L182

Reply:
If that's the only non-dict usage why not update that to produce a dict and make the code here simpler?

Reply:
I do not believe the code to handle this is overly (or even slightly) complicated. In addition to allowing `load_custom_wcs()` to return a simple WCS object, it also allows users who instantiate `ResampleImage` (or resample step) directly to pass directly a WCS object without having to put it into a dictionary.

Reply:
I don't feel the same way. Since the output of `load_custom_wcs` passes through the `ResampleImage` constructor in jwst why can't it wrap it in a dict and pass it to this class (which could support 1 less option for this input type which would certainly be simpler).

Reply:
For the `Step` interface in jwst I'm in favor of the second. The point I'm trying to make here is that the `Step` can take whatever the user provides (a `gwcs.WCS` instance or a filename pointing to an asdf file) and convert it to a single (or minimal set) of formats that the stcal interface implemented here can support. I don't see why stcal needs to support such a wide variety of formats when the step can convert to a smaller set of supported ones.